lib/services/outdoor_path_service.dart

```diff
@@ -0,0 +1,247 @@
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+import 'package:google_maps_util/google_maps_util.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:concordia_go/utilities/direction.dart';
+import 'package:concordia_go/utilities/segment.dart';
+import 'package:concordia_go/utilities/journey.dart';
+import 'package:flutter/material.dart';
+
+
+var _apiKey = '';
+
+class OutdoorPathService {
+  static final Set<Polyline> _polyLines = {};
+  static Journey _listDirections = Journey();
+  static List<Direction> _singleDirections = List<Direction>();
+  static int _currentInstruction = 0;
+
+  static void transitDirections(startLat, startLng, endLat, endLng, buildingDestination) async {
+    _singleDirections = List<Direction>();
+    _listDirections = Journey();
+    String url =
+        "https://maps.googleapis.com/maps/api/directions/json?origin=${startLat},${startLng}&destination=${endLat},${endLng}&mode=transit&key=${_apiKey}";
+    http.Response response = await http.get(url);
+    Map values = jsonDecode(response.body);
+    PolyUtil myPoints = PolyUtil();
+    for (int i = 0; i < values['routes'][0]['legs'][0]['steps'].length; i++) {
+      bool subInstruction = true;
+      var arrival_time = values['routes'][0]['legs'][0]['arrival_time']['text'];
+      var pointArray = myPoints.decode(values['routes'][0]['legs'][0]['steps'][i]['polyline']['points']);
+      Segment newSegment;
+      if (values['routes'][0]['legs'][0]['steps'][i]['travel_mode'] == 'WALKING') {
```

Comment:
@MoreWaqar

Reply:
They all have different paths in the JSON if you look at them all closely. It would note reduce the amount of lines or complexity. 

There is a possibility that it would look slightly cleaner in the form constant['steps][i]['travel_mode']

Reply:
Lets put them all in constants. It is the proper coding convention.  Duplicated string literals make the process of refactoring error-prone